src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/ExpressionEvaluatorFactory.cs

```diff
@@ -0,0 +1,52 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using DevLab.JmesPath;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class ExpressionEvaluatorFactory : IExpressionEvaluatorFactory
+    {
+        private JmesPath _jmesPath;
+
+        public ExpressionEvaluatorFactory()
+        {
+            _jmesPath = new JmesPath();
+
+            /*
+             * register additional custom functions.
```

Comment:
I'm thinking that we can create a separate Nuget to hold these functions. That way they could be shared across teams. Thoughts?

Reply:
I also like the idea of separating them out.

Reply:
Yes a separate library which will result in a separate package makes sense.  Long term we will want to migrate common code into the health shared repo.  The OSS project predates the existence of that repo and we haven't updated yet.